src/spec.rs

```diff
@@ -190,7 +190,7 @@ pub(crate) fn commit_ivk(
         // - sinsemilla::CommitDomain::short_commit calls extract_p_bottom, which replaces
         //   the identity (which has no affine coordinates) with 0. but Sinsemilla is
         //   defined using incomplete addition, and thus will never produce the identity.
-        .map(NonZeroPallasBase::guaranteed)
+        .and_then(NonZeroPallasBase::from_base)
```

Comment:
The comment also needs updating. Something like:
```rust
    // sinsemilla::CommitDomain::short_commit returns a value in range [0..q_P] ∪ {⊥}:
    // - sinsemilla::HashDomain::hash_to_point uses incomplete addition and returns a
    //   point in P* ∪ {⊥}.
    // - sinsemilla::CommitDomain::commit applies a final complete addition step and
    //   returns a point in P ∪ {⊥}.
    // - 0 is not a valid x-coordinate for any Pallas point.
    // - sinsemilla::CommitDomain::short_commit calls extract_p_bottom, which replaces
    //   the identity (which has no affine coordinates) with 0.
    //
    // Commit^ivk.Output is specified as [1..q_P] ∪ {⊥}, so we explicitly check for 0 and
    // map it to None. Note that we are collapsing this case (which is rejected by the
    // circuit) with ⊥ (which the circuit explicitly allows for efficiency); this is fine
    // because we don't want users of the `orchard` crate to encounter either case.
```